src/read/read_ref.rs

```diff
@@ -65,23 +62,28 @@ where
     /// Note that not all writes will be included with this read -- only those that have been
     /// refreshed by the writer. If no refresh has happened, or the map has been destroyed, this
     /// function returns `None`.
-    pub fn get<'a, Q: ?Sized>(&'a self, key: &'_ Q) -> Option<&'a Values<V, S>>
+    pub fn get<Q: ?Sized>(&self, key: &Q) -> Option<&Values<V, S>>
     where
         K: Borrow<Q>,
         Q: Hash + Eq,
     {
-        let inner = self.guard.as_ref()?;
-        if !inner.is_ready() {
-            return None;
-        }
-        inner.data.get(key).map(Values::user_friendly)
+        self.guard.data.get(key).map(Values::user_friendly)
     }
-
-    /// Returns true if the writer has destroyed this map.
+    
+    /// Returns a reference to the values corresponding to the key.
+    ///
+    /// The key may be any borrowed form of the map's key type, but `Hash` and `Eq` on the borrowed
+    /// form *must* match those for the key type.
     ///
-    /// See [`WriteHandle::destroy`].
-    pub fn is_destroyed(&self) -> bool {
-        self.guard.is_none()
+    /// Note that not all writes will be included with this read -- only those that have been
+    /// refreshed by the writer. If no refresh has happened, or the map has been destroyed, this
+    /// function returns `None`.
+    pub fn single<Q: ?Sized>(&'rh self, key: &'rh Q) -> Option<&'rh V>
```

Comment:
The lifetime on `key` should be `'_` to avoid making the return type tied to the the key's lifetime.